packages/dds/tree/src/shared-tree-core/transaction.ts

```diff
@@ -163,3 +172,140 @@ export class TransactionStack implements Transactor, IDisposable {
 		}
 	}
 }
+
+/**
+ * An implementation of {@link Transactor} that {@link TransactionStack | uses a stack} and a {@link SharedTreeBranch | branch} to manage transactions.
+ * @remarks Given a branch, this class will fork the branch when a transaction begins and squash the forked branch back into the original branch when the transaction ends.
+ * This class provides conveniences for interacting with the {@link SquashingTransactionStack.activeBranch | active branch} in a way that is stable across transaction boundaries.
+ * For example, see {@link SquashingTransactionStack.activeBranchEditor | activeBranchEditor} and {@link SquashingTransactionStack.activeBranchEvents | activeBranchEvents}.
+ */
+export class SquashingTransactionStack<
+	TEditor extends ChangeFamilyEditor,
+	TChange,
+> extends TransactionStack {
+	public readonly branch: SharedTreeBranch<TEditor, TChange>;
+	#transactionBranch?: SharedTreeBranch<TEditor, TChange>;
+
+	/**
+	 * An editor for whichever branch is currently the {@link SquashingTransactionStack.activeBranch | active branch}.
+	 * @remarks This editor can safely be held on to across transaction boundaries, as it will properly delegate to the correct branch.
+	 * In contrast, it is not safe to hold onto e.g. `activeBranch.editor` across transaction boundaries, since the active branch may change.
+	 */
+	public readonly activeBranchEditor = new Proxy<TEditor>({} as unknown as TEditor, {
```

Comment:
This technique isn't actually required at the moment because every usage of `checkout.editor` uses the editor immediately and then drops it (e.g. `checkout.editor.valueField(...).set(...);`). No usages hold on to the editor across edits, let alone across transactions. However, I thought it prudent to do this just to be sure.